tests/rptest/tests/pandaproxy_test.py

```diff
@@ -1860,3 +1861,63 @@ def test_mtls_and_basic_auth(self):
         result = result_raw.json()
         self.logger.debug(result)
         assert result[0] == self.topic
+
+
+class BasicAuthUpgradeTest(PandaProxyEndpoints):
+    topics = [
+        TopicSpec(),
+    ]
+
+    def __init__(self, context):
+        # Dissable pandaproxy by default since it is set later in the test
+        super(BasicAuthUpgradeTest, self).__init__(context,
+                                                   pandaproxy_config=None)
+
+        self.installer = self.redpanda._installer
+
+    def setUp(self):
+        # Do not call super's setUp yet because redpanda
+        # is manually started in the test methods
+        pass
+
+    def check_usage(self, security_feature: Optional[str] = None):
+        auth = None
+        if security_feature is not None and security_feature == 'http_basic':
+            super_username, super_password, _ = self.redpanda.SUPERUSER_CREDENTIALS
+            auth = (super_username, super_password)
+
+        result_raw = self._get_topics(auth=auth)
+        assert result_raw.status_code == requests.codes.ok
+        result = result_raw.json()
+        assert result[0] == self.topic
+
+    @cluster(num_nodes=3)
+    def test_upgrade_and_enable_basic_auth(self):
+        # Initial cluster does not have auth enabled
+        old_version, old_version_str = self.installer.install(
+            self.redpanda.nodes, (22, 2))
+        # Enable Pandaproxy but do not use client cache configs
+        pandaproxy_config = PandaproxyConfig()
+        pandaproxy_config.cache_keep_alive_ms = None
+        pandaproxy_config.cache_max_size = None
+        self.redpanda.set_pandaproxy_settings(pandaproxy_config)
+        self.redpanda.start()
+        self._create_initial_topics()
+        self.check_usage()
+
+        # Upgrade to cluster with basic auth support
+        # and test with basic auth enabled
+        self.installer.install(self.redpanda.nodes, (23, 1))
```

Comment:
Skipping versions (v22.3.x) isn't officially supported, and more recently (on dev/v23.2.x), explicitly rejected (assuming the logical version is bumped for each major release, which seems likely).

This test remains useful whilst v23.1.x is actively maintained, but does nothing to prevent regressions on new feature branches.

I don't know if we have any guidelines on how to write upgrade tests that remain relevant and won't become a maintenance burden. @jcsp any thoughts?